### ISSUE TITLE
Bug: Wrong AST for `import.defer()`

### ISSUE THREAD

**Issue by fisker** [typescript-eslint/typescript-eslint#12606]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.
- [ ] If I used AI assistance to prepare this issue, I have reviewed and followed the [AI Contribution Policy](https://typescript-eslint.io/contributing/ai-policy/).

### Relevant Package

typescript-estree

### Repro Code

```TypeScript
import.defer("foo")
```

### ESLint Config

```javascript
N/A
```

### tsconfig

```jsonc
N/A
```

### Expected Result

It should be an `ImportExpression` with `phase: "defer"`.

According to ESTree https://github.com/estree/estree/blob/875bf70440a8870c4a663865a7a41300cf1add55/stage3/defer-import-eval.md#importexpression

### Actual Result

`CallExpression` with `MetaProperty` as callee.

### Additional Info

I think we should also reject 

```
import.source('foo')
```

since we don't support

```
import source foo from "foo"
```

yet. https://github.com/tc39/proposal-source-phase-imports

And

```
import.unknown('foo')
```

But

```
import.meta('foo')
```

should be valid.

(Can open a separate issue for it)

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/typescript-estree` | `8.65.0` |

**Comment by kirkwaiblinger** [comment]
> ### Repro Code
> import.defer("foo")
>
> ### Expected Result
> It should be an `ImportExpression` with `phase: "defer"`.
> 
> According to ESTree https://github.com/estree/estree/blob/875bf70440a8870c4a663865a7a41300cf1add55/stage3/defer-import-eval.md#importexpression
> ### Actual Result
> `CallExpression` with `MetaProperty` as callee.

Seems straightforward but tagging @StyleShit, who did #12513, in case this was intentional rather than a simple oversight.

> ### Additional Info
> I think we should also reject
> 
> ```
> import.source('foo')
> ```
> 
> since we don't support
> 
> ```
> import source foo from "foo"
> ```
> 
> yet. https://github.com/tc39/proposal-source-phase-imports

This part I don't really follow... Why not let it keep being parsed as a MetaProperty until `import source` support lands in TS?

> And
> 
> ```
> import.unknown('foo')
> ```

Not sure what this is saying? 

> But
> 
> ```
> import.meta('foo')
> ```
> 
> should be valid.
> 
> (Can open a separate issue for it)

Maybe this would be helpful, since I'm having a hard time following what the second part is saying?

**Comment by fisker** [comment]
There are 3 issues

1) `import.defer('foo')` should be `ImportExpression`
2) Invalid meta property should be forbidden, only `import.meta` and `import.defer` are valid for now
3) `import.defer` must followed by `(`  (Not mentioned originally)

2 and 3 can be fixed separately.

**Comment by fisker** [comment]
> Why not let it keep being parsed as a MetaProperty until import source support lands in TS?

Because the AST will be wrong again. https://github.com/estree/estree/blob/875bf70440a8870c4a663865a7a41300cf1add55/stage3/source-phase-imports.md#importexpression

**Comment by kirkwaiblinger** [comment]
Oh, I understand now, I was tripping over the difference between `import.foo` and `import.meta.foo` 👍 

Yes, I think per the spec it makes sense to make only consider `import.meta` be `MetaProperty` and not just any `import.*`

https://github.com/estree/estree/blob/875bf70440a8870c4a663865a7a41300cf1add55/es2020.md?plain=1#L221-L223

Thanks for explaining!

**Comment by StyleShit** [comment]
> ... in case this was intentional rather than a simple oversight

This is definitely an oversight, idk how I missed that. Thanks for reporting!
